### PR TITLE
Clamp resolution to 5 microseconds

### LINKED ISSUE #4: Timing attacks
We’ve recently published a report describing a new Javascript-based network attack. Our attack lets a malicious website learn a surprising amount of personal information about an innocent user, and is largely based upon repeated calls to performance.now() as a measurement method.  We have a proof-of-concept demo for this attack which I can share off-list with relevant stakeholders.  A draft report is also available online here:
http://arxiv.org/abs/1502.07373 <http://arxiv.org/abs/1502.07373>

From our testing it is very apparent that somewhat reducing the resolution of performance.now() will make the attack much more difficult. We noticed that various browser vendors implement the call with different precisions, with Firefox for Linux and MacOS going down to the single nanosecond. Our attack as described becomes very difficult to launch once the resolution of performance.now() is upper-bounded to 5 thousandths of a millisecond.  

I would like to suggest that you change the spec to recommend this countermeasure.  In your opinion, does this modification incur too much of a usability loss (to games, music, VR, etc)?

### PR DESCRIPTION
**Pull request by igrigorik**
Closes #4.

### CODE REVIEW

**Comment by toddreifsteck** [comment]
LGTM

File: index.html [hunk 1]
```diff
@@ -303,11 +302,7 @@ <h3 id="sec-DOMHighResTimeStamp"><span class="secno">4.3 </span>The <code>DOMHig
 
 <h3>Privacy and Security</h3>
 <p>
-Statistical fingerprinting is a privacy concern where a malicious web site may determine whether a user has visited a
-third-party web site by measuring the timing of cache hits and misses of resources in the third-party web site.
-Though the <a>now</a> method of the <a>Performance</a> interface
-returns time data to a greater accuracy than before, it does not make this privacy concern significantly worse than it was already.
-</p>
+Statistical fingerprinting is a privacy concern where a malicious web site may use high resolution timing data of various browser or application-initiated operations to identify a particular user. To mitigate such attacks, the recommended minimum resolution of the <a>Performance</a> interface should be set to 5 microseconds.</p>
```
Comment (yoavweiss) [inline index.html]: Do we want to link to the [original paper](http://arxiv.org/pdf/1502.07373v2.pdf) here?
Also - it permits way more than privacy/fingerprinting concerns. AFAIU it enables to estimate CPU cache contents and deduce (or narrow down range of) passwords, secret keys, etc.

Comment (plehegar) [inline index.html]: +1 to linking to the original paper (see navigationg-timing for an example on how to add a local biblio). I also suggest adding the paper authors in the acknowledgments section.

**Comment by igrigorik** [comment]
@yoavweiss @plehegar updated, ptal. I'll have to rebase against latest changes before merging.

**Comment by plehegar** [comment]
Sorry, one more change I think, we have a sentence in the spec that says:
"A DOMHighResTimeStamp SHOULD represent a time in milliseconds accurate to a microsecond."
This would be need to be updated as well.

**Comment by igrigorik** [comment]
Looking for mentions of "accurate to": 
- https://github.com/w3c/hr-time/blob/time-resolution/index.html#L254
- https://github.com/w3c/hr-time/blob/time-resolution/index.html#L257 (the second is for millisecond accuracy)

Am I missing it?

**Comment by plehegar** [comment]
LGTM, merging.

**Pull request merged**